FlowCrypt/Extensions/URLSessionExtension.swift

```diff
@@ -18,19 +18,27 @@ struct HttpErr: Error {
 
 extension URLSession {
 
-    func call(_ urlRequest: URLRequest, tolerateStatus: [Int]? = nil) -> Promise<HttpRes> { return Promise { resolve, reject in
-        let start = DispatchTime.now()
-        self.dataTask(with: urlRequest) { data, response, error in
-            let res = response as? HTTPURLResponse
-            let status = res?.statusCode ?? -1
-            print("URLSession.call status:\(status) ms:\(start.millisecondsSince()) \(urlRequest.url?.absoluteString ?? "??")")
-            if error == nil && data != nil && (200...299 ~= status || tolerateStatus?.firstIndex(of: status) != nil) {
-                resolve(HttpRes(status: status, data: data!))
-            } else {
-                reject(HttpErr(status: status, data: data, error: error))
-            }
-        }.resume()
-    }}
+    func call(_ urlRequest: URLRequest, tolerateStatus: [Int]? = nil) -> Promise<HttpRes> {
+        return Promise { resolve, reject in
+            let start = DispatchTime.now()
+            self.dataTask(with: urlRequest) { data, response, error in
+                let res = response as? HTTPURLResponse
+                let status = res?.statusCode ?? Constants.Global.generalError
+
+                print("URLSession.call status:\(status) ms:\(start.millisecondsSince()) \(urlRequest.url?.absoluteString ?? "??")")
+
+                let validStatusCode = 200...299
+                let isCodeVaild = validStatusCode ~= status || tolerateStatus?.contains(status)
+                let isValidResonse = error == nil && isCodeVaild
+
+                if let data = data, isValidResonse {
+                    resolve(HttpRes(status: status, data: data!))
```

Comment:
My xcode is showing many different errors across the codebase now, for example:
<img width="1298" alt="image" src="https://user-images.githubusercontent.com/6306961/63372028-dd418180-c3b7-11e9-88e5-e25a6c33e8e8.png">

Should I update some settings, etc?